### PR TITLE
Additional tests and fixes for SOQL in Snowfakery

### PR DESCRIPTION
**Pull request by prescod**
Additional tests and fixes for SOQL in Snowfakery

### CODE REVIEW

File: requirements/prod.in [hunk 1]
```diff
@@ -1,5 +1,4 @@
 contextvars
-cumulusci
```
Comment (Br4nd0R) [inline requirements/prod.in:2]: No more hard dependency on CCI?

File: snowfakery/standard_plugins/Salesforce.py [hunk 2]
```diff
@@ -114,17 +109,23 @@ def compose_query(self, context_name, **kwargs):
 
     @staticmethod
     def _get_sf_clients(orgname):
+
         try:
+            from cumulusci.cli.runtime import CliRuntime
```
Comment (Br4nd0R) [inline snowfakery/standard_plugins/Salesforce.py:114]: I thought you removed the dependency on CumulusCI? Is it present in certain contexts?

File: requirements/prod.in (hunk 1 above)
Comment (prescod) [reply requirements/prod.in:2]: Correct.

File: snowfakery/standard_plugins/Salesforce.py (hunk 2 above)
Comment (prescod) [reply snowfakery/standard_plugins/Salesforce.py:114]: Removed the hard (build-time, load-time) dependency. It still depends on CumulusCI if the user uses particular features.

File: tests/conftest_extras_w_cci.py [hunk 3]
```diff
@@ -62,7 +63,29 @@ def no_services(*args, **kwargs):
         yield fallback_org_config
 
 
-vcr_config = fixture(cci_vcr_config, scope="module")
+# TODO: Port this back to CCI
+def sf_before_record_response(response):
+    # salesforce_bulk needs the Content-Type header.
+    print(response["headers"])
```
Comment (Br4nd0R) [inline tests/conftest_extras_w_cci.py:69]: Do you want to include this, or is this leftover from debugging?

Comment (prescod) [reply tests/conftest_extras_w_cci.py:69]: Debugging. I'll take it out.

**Review by Br4nd0R — APPROVED** [review]
Only a couple questions. This looks good to me :)

**Review by Br4nd0R — APPROVED** [review]

**Pull request merged**